File: 15-07-2024/Palindrome-String/Solutions/Solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        sb = []
        for char in s:
            if char.isalnum():
                if char.isupper():
                    sb.append(char.lower())
                else:
                    sb.append(char)
        
        i, j = 0, len(sb) - 1
        while i < j:
            if sb[i] == sb[j]:
                i += 1
                j -= 1
            else:
                return False
        return True

File: 15-07-2024/Palindrome-String/Solutions/Solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string sb;
        for (int i = 0; i < s.length(); ++i) {
            if ((s[i] >= 'A' && s[i] <= 'Z') || (s[i] >= 'a' && s[i] <= 'z') || (s[i] >= '0' && s[i] <= '9')) {
                if (s[i] >= 'A' && s[i] <= 'Z') {
                    sb += (char)(s[i] + 32);
                } else {
                    sb += s[i];
                }
            }
        }

        int i = 0, j = sb.length() - 1;
        while (i < j) {
            if (sb[i] == sb[j]) {
                ++i;
                --j;
            } else {
                return false;
            }
        }
        return true;
    }
};